onnx/defs/function.cc

```diff
@@ -105,6 +140,14 @@ void FunctionExpandHelper(
       }
     }
   }
+
+  // Remove function node from graph
+  for (int i = 0; i < g.node().size(); ++i) {
```

Comment:
can this be done when doing in-lining above? this removal may be not effective if the model is very large.

Reply:
@linkerzhang This part itself cannot be done in the in-lining above as it's currently written.

AFAICT, current in-lining is based on adding all the nodes of the expanded subgraph to the end of the main graph's node list:

https://github.com/onnx/onnx/blob/b8966dc607b4a4f495c68c2c44996cada4c2c843/onnx/defs/function.cc#L79-L80

The graph this approach generates is **not sorted** topologically.

We could try to expand this PR, to replace the function node with the expanded subgraph (instead of deleting it after the expansion).